src/test_utils/katana/genesis.rs

```diff
@@ -354,13 +354,11 @@ impl KatanaGenesisBuilder<Initialized> {
 }
 
 fn compute_class_hash(class: &Value) -> Result<FieldElement> {
-    match serde_json::from_value::<SierraClass>(class.clone()) {
-        Ok(sierra) => Ok(sierra.class_hash()?),
-        Err(_) => {
-            let casm: LegacyContractClass =
-                serde_json::from_value(class.clone()).expect("Failed to parse class code v0");
-            Ok(casm.class_hash()?)
-        }
+    if let Ok(sierra) = serde_json::from_value::<SierraClass>(class.clone()) {
```

Comment:
so clippy favous `if let else` over `match` for Result?

Reply:
Yes related to this one: https://rust-lang.github.io/rust-clippy/master/index.html#/single_match_else